src/panoptes/pocs/focuser/astromechanics.py

```diff
@@ -0,0 +1,266 @@
+        hitting a stop.
+        """
+        self._is_moving = True
+        try:
+            ini_pos = self._send_command("P#", response_length=1)[0].replace("#", "")
+            new_pos = int(ini_pos) + increment
+            self.move_to(new_pos)
+        finally:
+            # Focuser move commands block until the move is finished, so if the command has
+            # returned then the focuser is no longer moving.
+            self._is_moving = False
+
+        self.logger.debug("Moved by {} encoder units. Current position is {}".format(increment, new_pos))
+        return new_pos
+
+    ##################################################################################################
+    # Private Methods
+    ##################################################################################################
+
+    def _send_command(self, command, response_length=None):
+        """
+        Sends a command to the Focuser adaptor and retrieves the response.
+
+        Args:
+            command (string): command string to send (without newline), e.g. 'fa1000', 'pf'
+            response length (integer, optional, default=None): number of lines of response expected.
+                For most commands this should be 0 or 1. If None readlines() will be called to
+                capture all responses. As this will block until the timeout expires it should only
+                be used if the number of lines expected is not known (e.g. 'ds' command).
+
+        Returns:
+            list: possibly empty list containing the '\n' terminated lines of the response from the
+                adaptor.
+        """
+        if not self.is_connected:
+            self.logger.critical("Attempt to send command to {} when not connected!".format(self))
+            return
+
+        # Clear the input buffer in case there's anything left over in there.
+        self._serial_port.reset_input_buffer()
+
+        # Send command
+        self._serial_io.write(command + '\n')
+
+        # Depending on which command was sent there may or may not be any further
+        # response.
+        response = []
+
+        if response_length == 0:
+            # Not expecting any further response. Should check the buffer anyway in case an error
+            # message has been sent.
+            if self._serial_port.in_waiting:
+                response.append(self._serial_io.readline())
+
+        elif response_length > 0:
+            # Expecting some number of lines of response. Attempt to read that many lines.
+            for i in range(response_length):
+                response.append(self._serial_io.readline())
+
+        else:
+            # Don't know what to expect. Call readlines() to get whatever is there.
+            response.append(self._serial_io.readlines())
+
+        return response
+
+    def _initialise(self):
+        # Get initial position of focuser adaptor.
+        self._get_initial_position()
```

Comment:
I think you don't need `_get_initial_position`:
```suggestion
        self.logger.debug(f'Initial position of focuser is at {self.position} encoder units')
```